signalfx/resource_signalfx_team.go

```diff
@@ -0,0 +1,409 @@
+					ValidateFunc: validateNotification,
+				},
+				Description: "List of notification destinations to use for the major alerts category.",
+			},
+			"notifications_minor": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validateNotification,
+				},
+				Description: "List of notification destinations to use for the minor alerts category.",
+			},
+			"notifications_warning": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validateNotification,
+				},
+				Description: "List of notification destinations to use for the warning alerts category.",
+			},
+			"url": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "URL of the team",
+			},
+		},
+
+		Create: teamCreate,
+		Read:   teamRead,
+		Update: teamUpdate,
+		Delete: teamDelete,
+		Exists: teamExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+	}
+}
+
+/*
+  Use Resource object to construct json payload in order to create a team
+*/
+func getPayloadTeam(d *schema.ResourceData) (*team.CreateUpdateTeamRequest, error) {
+	t := &team.CreateUpdateTeamRequest{
+		Name:        d.Get("name").(string),
+		Description: d.Get("description").(string),
+	}
+
+	var members []string
+	if val, ok := d.GetOk("members"); ok {
+		tfValues := val.(*schema.Set).List()
+		for _, v := range tfValues {
+			members = append(members, v.(string))
+		}
+	}
+	t.Members = members
+
+	if val, ok := d.GetOk("notifications_critical"); ok {
+		nots, err := getNotificationList(val.([]interface{}))
+		if err != nil {
+			return t, err
+		}
+		t.NotificationLists.Critical = nots
+	}
+	if val, ok := d.GetOk("notifications_default"); ok {
+		nots, err := getNotificationList(val.([]interface{}))
+		if err != nil {
+			return t, err
+		}
+		t.NotificationLists.Default = nots
+	}
+	if val, ok := d.GetOk("notifications_info"); ok {
+		nots, err := getNotificationList(val.([]interface{}))
+		if err != nil {
+			return t, err
+		}
+		t.NotificationLists.Info = nots
+	}
+	if val, ok := d.GetOk("notifications_major"); ok {
+		nots, err := getNotificationList(val.([]interface{}))
+		if err != nil {
+			return t, err
+		}
+		t.NotificationLists.Major = nots
+	}
+	if val, ok := d.GetOk("notifications_minor"); ok {
+		nots, err := getNotificationList(val.([]interface{}))
+		if err != nil {
+			return t, err
+		}
+		t.NotificationLists.Minor = nots
+	}
+	if val, ok := d.GetOk("notifications_warning"); ok {
+		nots, err := getNotificationList(val.([]interface{}))
+		if err != nil {
+			return t, err
+		}
+		t.NotificationLists.Warning = nots
+	}
+
+	return t, nil
+}
+
+// Convert the list of TF data into proper objects
+func getNotificationList(items []interface{}) ([]*notification.Notification, error) {
+	if len(items) < 0 {
```

Comment:
`<= 0`?  Pretty sure len can never be less than zero.

Reply:
How'd I end up with that!? :)